### ISSUE TITLE
Improve startup error message, include "dnGrep" in the message

### ISSUE THREAD

**Issue by daniel3550** [dnGrep/dnGrep#500]
I encountered #467 but unfortunately it took me a while to figure out dnGrep was causing the problem. The windows dialog that pops up just has a generic error message without the program name. 

![image](https://user-images.githubusercontent.com/8118451/114249602-b7883800-9960-11eb-8141-9dd964cce074.png)

**Comment by daniel3550** [comment]
Actually I uninstalled, deleted C:\Program Files\dnGREP, restarted machine, installed dnGrep, and I get the error again after restarting. So maybe this is a new issue. 


```
<logevent time="2021-04-09 18:23:04.7717" level="Error" logger="dnGREP.WPF.App">
  <message>Unhandled exception caught</message>
  <exception>System.ArgumentException: &apos;-511&apos; is not a valid value for property &apos;Width&apos;.
   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)
   at System.Windows.DependencyObject.SetValue(DependencyProperty dp, Object value)
   at DockFloat.Extensions.ConstrainToScreen(Window window)
   at dnGREP.WPF.MainForm.&lt;&gt;c__DisplayClass5_0.&lt;.ctor&gt;b__0(Object s, RoutedEventArgs e)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)
   at System.Windows.BroadcastEventHelper.BroadcastLoadedEvent(Object root)
   at MS.Internal.LoadedOrUnloadedOperation.DoWork()
   at System.Windows.Media.MediaContext.FireLoadedPendingCallbacks()
   at System.Windows.Media.MediaContext.FireInvokeOnRenderCallbacks()
   at System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   at System.Windows.Media.MediaContext.RenderMessageHandler(Object resizedCompositionTarget)
   at System.Windows.Interop.HwndTarget.OnResize()
   at System.Windows.Interop.HwndTarget.HandleMessage(WindowMessage msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Interop.HwndSource.HwndTargetFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean&amp; handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean&amp; handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)</exception>
</logevent>
```

**Comment by doug24** [comment]
The error you are getting is the code is trying to set the width of the dnGrep main window so it is contained on one display screen.  I can't tell why, but the code has failed to get the correct screen bounds for the display where dnGrep is started.   The code should have an error check here to ensure it doesn't try to make the window width a negative number, and I will fix that.

What version of Windows are you running?
How many monitors do you have and how are they positioned relative to the main/default monitor?

You may be able to get dnGrep running by editing the dnGrep.Settings.dat file in:
C:\Users\_user_\AppData\Roaming\dnGREP

Change the start position and size of the main window by editing this property:
<item key="MainWindowBounds">10,10,970,725</item>
the values are:  >X, Y, Width, Height<

You must use dot for the decimal separator and comma for the item separator

**Comment by daniel3550** [comment]
Windows 10, Version 20H2, Build 19042.746
2 identical monitors, extended, 2nd monitor is on right of primary

Note dnGrep starts fine when I manually launch it. I only get the error during windows startup. I do not see a line in my settings file as you describe

C:\Users\[USER]\AppData\Roaming\dnGrep\dnGREP.Settings.dat

```
<?xml version="1.0" encoding="utf-8"?>
<dictionary>
  <item key="SearchFor">does not</item>
  <item key="ReplaceWith"></item>
  <item key="FilePattern">*</item>
  <item key="FilePatternIgnore"></item>
  <item key="PreviewWindowWrap">AAEAAAD/////AQAAAAAAAAAEAQAAAA5TeXN0ZW0uQm9vbGVhbgEAAAAHbV92YWx1ZQABAAs=</item>
  <item key="PreviewWindowFont">12</item>
  <item key="FastSearchBookmarks">AAEAAAD/////AQAAAAAAAAAEAQAAAH9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5MaXN0YDFbW1N5c3RlbS5TdHJpbmcsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAwAAAAZfaXRlbXMFX3NpemUIX3ZlcnNpb24GAAAICAkCAAAAFAAAABQAAAARAgAAACAAAAAGAwAAAAhkb2VzIG5vdAYEAAAADGludGVyZmFjZSBvbgYFAAAACUNvdWxkIG5vdAYGAAAADUNvdWxkIG5vdCBzZXQGBwAAAB9Db3VsZCBub3Qgc2V0IGNvbnRyb2wgaW50ZXJmYWNlBggAAAAIUHJvdG9jb2wGCQAAABpQcm90b2NvbCAuKiBkb2VzIG5vdCBleGlzdAYKAAAAGGhhbF9oaWZfc3ZjX3Rlc3RfcmVzdWx0cwYLAAAAB2hhbF9oaWYGDAAAAAdkZGFyZGVuBg0AAAADRU1VBg4AAAADZW11Bg8AAAAEZW11LgYQAAAACXdyaXRlX2JpbgYRAAAACnN0dWRpb2xpbmsGEgAAAA1ibGFja0JveF9yZWFkBhMAAAAIYmxhY2tCb3gGFAAAAAYweDFiMDgGFQAAAAYweDFCMDgGFgAAAAxyZWFkV3JpdGVSYXcNDAs=</item>
  <item key="FastReplaceBookmarks">AAEAAAD/////AQAAAAAAAAAEAQAAAH9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5MaXN0YDFbW1N5c3RlbS5TdHJpbmcsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAwAAAAZfaXRlbXMFX3NpemUIX3ZlcnNpb24GAAAICAkCAAAAAwAAAAMAAAARAgAAAAQAAAAGAwAAAAAGBAAAAANDREIGBQAAAANjZGIKCw==</item>
  <item key="FastFileMatchBookmarks">AAEAAAD/////AQAAAAAAAAAEAQAAAH9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5MaXN0YDFbW1N5c3RlbS5TdHJpbmcsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAwAAAAZfaXRlbXMFX3NpemUIX3ZlcnNpb24GAAAICAkCAAAABAAAAAQAAAARAgAAAAQAAAAGAwAAAAEqBgQAAAAEKi5weQYFAAAACCphcHgqLnB5BgYAAAADKi4qCw==</item>
  <item key="FastFileNotMatchBookmarks">AAEAAAD/////AQAAAAAAAAAEAQAAAH9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5MaXN0YDFbW1N5c3RlbS5TdHJpbmcsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAwAAAAZfaXRlbXMFX3NpemUIX3ZlcnNpb24GAAAICAkCAAAABQAAAAUAAAARAgAAAAgAAAAGAwAAAAAGBAAAACMqLmF4ZiwgKi5odG0sIE1haG9uZXlfMC45LjAqLCAqLmRsbAYFAAAAHCouYXhmLCAqLmh0bSwgTWFob25leV8wLjkuMCoGBgAAACYqLmF4ZiwgKi5odG0sIE1haG9uZXlfMC45LjAqLCBtYWhvbmV5KgYHAAAAGyouYXhmLCAqLmh0bSwgTWFob25leV8wLjkuMA0DCw==</item>
  <item key="FastPathBookmarks">AAEAAAD/////AQAAAAAAAAAEAQAAAH9TeXN0ZW0uQ29sbGVjdGlvbnMuR2VuZXJpYy5MaXN0YDFbW1N5c3RlbS5TdHJpbmcsIG1zY29ybGliLCBWZXJzaW9uPTQuMC4wLjAsIEN1bHR1cmU9bmV1dHJhbCwgUHVibGljS2V5VG9rZW49Yjc3YTVjNTYxOTM0ZTA4OV1dAwAAAAZfaXRlbXMFX3NpemUIX3ZlcnNpb24GAAAICAkCAAAADgAAAA4AAAARAgAAABAAAAAGAwAAAFRDOlxVc2Vyc1xkZGFyZGVuXEFwcERhdGFcTG9jYWxcUHJvZ3JhbXNcUHl0aG9uXFB5dGhvbjM3XExpYlxzaXRlLXBhY2thZ2VzXHN0dWRpb2xpbmsGBAAAAGtDOlxVc2Vyc1xkZGFyZGVuXFdvcmtcUHJvamVjdHNcTWFob25leVxtYWhvbmV5X2NzMzdsMTBfdGVtcFxtYWhvbmV5X2NzMzdsMTBcY2RiXENTMzdMMTBfU0NTXHNjc2xpbmtfc2NyaXB0cwYFAAAAJ0M6XFVzZXJzXGRkYXJkZW5cRGVza3RvcFx0ZW1wXHNwaGluY3RlcgYGAAAATUM6XFVzZXJzXGRkYXJkZW5cRGVza3RvcFx0ZW1wXG1haG9uZXlfY3MzN2wxMFxjZGJcQ1MzN0wxMF9TQ1Ncc2NzbGlua19zY3JpcHRzBgcAAAA9QzpcVXNlcnNcZGRhcmRlblxEb3dubG9hZHNcTWFob25leSAoMTQpXEN1c3RvbWVyX0Rpc3RyaWJ1dGlvbgYIAAAAMUM6XFVzZXJzXGRkYXJkZW5cV29ya1xQcm9qZWN0c1xNYWhvbmV5XGZ3XG1haG9uZXkGCQAAAD1DOlxVc2Vyc1xkZGFyZGVuXERlc2t0b3BcdGVtcFxtYWhvbmV5X2NzMzdsMTBcZW11XENTMzdMMTBfU0NTBgoAAAAfQzpcVXNlcnNcZGRhcmRlblxXb3JrXFByb2plY3RzXAYLAAAAQEM6XFVzZXJzXGRkYXJkZW5cV29ya1xQcm9qZWN0c1xjaXJydXN2dWVccHJvamVjdHNccmVlZFx1dGlsaXRpZXMGDAAAAFlDOlxVc2Vyc1xkZGFyZGVuXFdvcmtcY3J1c3B5XHB5dGhvbi0zLjcuNi4xMy13aW4tZW1iZWQtYW1kNjRcTGliXHNpdGUtcGFja2FnZXNcc3R1ZGlvbGluawYNAAAAK0M6XFVzZXJzXGRkYXJkZW5cV29ya1xQcm9qZWN0c1xTdGVybGluZ1xTQ1MGDgAAAFlDOlxVc2Vyc1xkZGFyZGVuXFdvcmtcUHJvamVjdHNcTWFob25leVxtYWhvbmV5X2NzMzdsMTBcZW11XENTMzdMMTBfU0NTXEJTUFx1dGlsc1xpbnRlbGhleAYPAAAAKkM6XFVzZXJzXGRkYXJkZW5cU291bmRDbGVhclN0dWRpb1x2MS43XGxvZwYQAAAAKUM6XFVzZXJzXGRkYXJkZW5cV29ya1xQcm9qZWN0c1xNYWhvbmV5XGZ3DQIL</item>
  <item key="SortDirection">Ascending</item>
  <item key="TypeOfSort">FileNameDepthFirst</item>
  <item key="ShowResultOptions">True</item>
  <item key="ResultsTreeScale">1</item>
  <item key="HighlightMatches">True</item>
  <item key="ShowLinesInContext">True</item>
  <item key="ContextLinesBefore">2</item>
  <item key="ContextLinesAfter">3</item>
  <item key="IncludeHidden">True</item>
  <item key="IncludeBinary">True</item>
  <item key="IncludeArchive">False</item>
  <item key="SearchParallel">True</item>
  <item key="IncludeSubfolder">True</item>
  <item key="MaxSubfolderDepth">-1</item>
  <item key="TypeOfSearch">Regex</item>
  <item key="CodePage">-1</item>
  <item key="TypeOfFileSearch">Asterisk</item>
  <item key="UseGitignore">True</item>
  <item key="UseFileSizeFilter">No</item>
  <item key="CaseSensitive">False</item>
  <item key="Multiline">False</item>
  <item key="Singleline">False</item>
  <item key="StopAfterFirstMatch">False</item>
  <item key="WholeWord">False</item>
  <item key="BooleanOperators">False</item>
  <item key="HighlightCaptureGroups">False</item>
  <item key="SizeFrom">0</item>
  <item key="SizeTo">100</item>
  <item key="IsFiltersExpanded">False</item>
  <item key="PreviewFileContent">False</item>
  <item key="UseFileDateFilter">None</item>
  <item key="TypeOfTimeRangeFilter">None</item>
  <item key="StartDate">AAEAAAD/////AQAAAAAAAAAEAQAAAA9TeXN0ZW0uRGF0ZVRpbWUCAAAABXRpY2tzCGRhdGVEYXRhAAAJEAAAAAAAAAAAAAAAAAAAAAAL</item>
  <item key="EndDate">AAEAAAD/////AQAAAAAAAAAEAQAAAA9TeXN0ZW0uRGF0ZVRpbWUCAAAABXRpY2tzCGRhdGVEYXRhAAAJEAAAAAAAAAAAAAAAAAAAAAAL</item>
  <item key="HoursFrom">0</item>
  <item key="HoursTo">8</item>
  <item key="ApplicationFontFamily">Segoe UI</item>
  <item key="MainFormFontSize">12</item>
  <item key="ReplaceFormFontSize">12</item>
  <item key="DialogFontSize">12</item>
  <item key="SearchFolder">C:\Users\ddarden\AppData\Local\Programs\Python\Python37\Lib\site-packages\studiolink</item>
  <item key="LastCheckedVersion">AAEAAAD/////AQAAAAAAAAAEAQAAAA9TeXN0ZW0uRGF0ZVRpbWUCAAAABXRpY2tzCGRhdGVEYXRhAAAJEC+xT3+C+9gIL7FPf4L72IgL</item>
  <item key="FollowSymlinks">False</item>
  <item key="ReplaceWindowWrap">AAEAAAD/////AQAAAAAAAAAEAQAAAA5TeXN0ZW0uQm9vbGVhbgEAAAAHbV92YWx1ZQABAAs=</item>
  <item key="ReplaceWindowFontSize">14</item>
  <item key="PreviewBounds">0,0,0,0</item>
  <item key="PreviewWindowState">Normal</item>
  <item key="PreviewDocked">True</item>
  <item key="PreviewDockSide">Right</item>
  <item key="PreviewDockedWidth">796.783870186313</item>
  <item key="PreviewDockedHeight">295.833333333333</item>
  <item key="PreviewHidden">False</item>
  <item key="PreviewAutoPosition">True</item>
  <item key="CaptureGroupSearch">False</item>
  <item key="MainWindowBounds">2431,216,1936,1176</item>
  <item key="MainWindowState">Maximized</item>
</dictionary>
```

**Comment by doug24** [comment]
Thanks for the update.  I edited my comment, the markdown editor hide the xml notation.   I'll fix the message and add the value check.

**Comment by doug24** [comment]
Fixed in Release 2.9.345
